nuget/lib/dependabot/nuget/update_checker/nupkg_fetcher.rb

```diff
@@ -111,7 +111,13 @@ def self.fetch_stream(stream_url, auth_header, max_redirects = 5)
         current_redirects = 0
 
         loop do
-          response = fetch_url_with_auth(current_url, auth_header)
```

Comment:
`fetch_url_with_auth` was caching the result, but apparently repeatedly passing the result to `Zip::File.open_buffer()` will result in a corrupted buffer, so the fix is to always download and to not cache anything.